### PR TITLE
fix(outdated): error when there are no config files

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit changes "deno outdated" subcommand to
error out if run in a directory that has no config file 
(including parent directories). This matches
"pnpm" behavior.

Also added tests for filtering that yields no results,
to ensure that it exists cleanly, that also matches "pnpm"
behavior.

Closes https://github.com/denoland/deno/issues/27287

### CODE REVIEW

File: cli/tools/registry/pm/outdated.rs [hunk 1]
```diff
@@ -179,6 +180,14 @@ pub async fn outdated(
   let jsr_fetch_resolver =
     Arc::new(JsrFetchResolver::new(file_fetcher.clone()));
 
+  if cli_options.maybe_lockfile().is_none() {
```
Comment (bartlomieju) [inline cli/tools/registry/pm/outdated.rs:183]: This is wrong, first check if there's either `deno.json` or `package.json`.

File: tests/specs/update/no_lockfile/__test__.jsonc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "tempDir": true,
+  "args": "outdated",
+  "exitCode": 1,
+  "output": "error: No lockfile in \"[WILDCARD]\". Run deno install to generate one.\n"
```
Comment (dsherret) [inline tests/specs/update/no_lockfile/__test__.jsonc:5]:
```suggestion
  "output": "error: No lockfile in \"[WILDLINE]\". Run deno install to generate one.\n"
```

File: tests/specs/update/no_config_file/__test__.jsonc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "tempDir": true,
+  "args": "outdated",
+  "exitCode": 1,
+  "output": "error: No deno.json or package.json in \"[WILDCARD]\".\n"
```
Comment (dsherret) [inline tests/specs/update/no_config_file/__test__.jsonc:5]: Nitpick:

```suggestion
  "output": "error: No deno.json or package.json in \"[WILDLINE]\".\n"
```

File: cli/tools/registry/pm/outdated.rs [hunk 4]
```diff
@@ -179,6 +180,31 @@ pub async fn outdated(
   let jsr_fetch_resolver =
     Arc::new(JsrFetchResolver::new(file_fetcher.clone()));
 
+  if !cli_options.start_dir.has_deno_json()
+    && !cli_options.start_dir.has_pkg_json()
+  {
+    bail!(
+      "No deno.json or package.json in {:?}.",
+      cli_options.initial_cwd(),
```
Comment (dsherret) [inline cli/tools/registry/pm/outdated.rs:188]: Nitpick to not rely on debug formatting:

```suggestion
      "No deno.json or package.json in \"{}\".",
      cli_options.initial_cwd().display(),
```

File: cli/tools/registry/pm/outdated.rs [hunk 5]
```diff
@@ -179,6 +180,31 @@ pub async fn outdated(
...
+  if !lockfile_exists {
+    bail!(
+      "{}",
+      color_print::cformat!(
+        "No lockfile in {:?}. Run <u>{}</> to generate one.",
+        cli_options.initial_cwd(),
+        "deno install"
+      )
+    );
+  }
```
Comment (dsherret) [inline cli/tools/registry/pm/outdated.rs:206]: What happens when someone has `"lock": false`? I feel like this would break them?

**Review by dsherret — COMMENTED** [review]
Looks almost good to me. I'm not sure about the lockfile check.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**